### PR TITLE
[LOOP-28] Upload loop settings data to Tidepool backend

### PR DESCRIPTION
**Pull request by darinkrauss**
- https://tidepool.atlassian.net/browse/LOOP-28
- Add TControllerSettingsDatum
- Add TPumpSettingsOverrideDeviceEventDatum
- Add fields to TCGMSettingsDatum
- Update fields of TPumpSettingsDatum
- Remove TApplicationSettingsDatum as unnecessary
- Add insulin formulation to bolus types
- Add tests

### CODE REVIEW

**Comment by darinkrauss** [comment]
LW PR for build: https://github.com/tidepool-org/LoopWorkspace/pull/698

File: TidepoolKit/TLogging.swift [hunk 1]
```diff
@@ -42,7 +42,7 @@ public protocol TLogging {
 public class TSharedLogging {
 
     /// A shared, global instance of TLogging.
-    public static var instance: TLogging?
+    public static weak var instance: TLogging?
```
Comment (rickpasetto) [inline TidepoolKit/TLogging.swift]: Who is supposed to set this?  Seems like an odd contract to have a singleton that doesn't hold a ref to this.  Now, anyone setting this like this:
```
class MyLogging: TLogging { ... }
TSharedLogging.instance = MyLogging()
```

will wind up losing their Logger.

**Review by rickpasetto — COMMENTED** [review]
Question about the contract regarding who should own the `TLogging` singleton.

File: TidepoolKit/TLogging.swift (hunk 1 above)
Comment (darinkrauss) [reply TidepoolKit/TLogging.swift]: Fixed. Logger is now a simple property on the `TAPI` object.

**Review by rickpasetto — APPROVED** [review]
👍

**Pull request merged**